spec/requests/odk_form_download_spec.rb

```diff
@@ -85,6 +87,54 @@
           end
         end
       end
+
+      context "for forms with audio prompts" do
+        let(:form) { create(:form, :published, mission: mission, question_types: %w[text integer]) }
+
+        before do
+          form.c[0].question.update!(audio_prompt: audio_fixture("powerup.mp3"))
+          form.c[1].question.update!(audio_prompt: audio_fixture("powerup.wav"))
+        end
+
+        it "should render manifest tags correctly" do
+          get("/m/#{mission.compact_name}/forms/#{form.id}/manifest")
+          expect(response).to be_success
+
+          download_url = "http://www.example.com/en/m/#{mission.compact_name}/questions"
+
+          assert_select("mediaFile", count: 2) do |elements|
+            assert_select(elements[0], "filename", text: "#{form.c[0].question.id}_audio_prompt.mp3")
+            assert_select(elements[0], "hash", text: "e7fe3aa406b8b67209b9d89c0cd50aa8")
+            assert_select(
+              elements[0],
+              "downloadUrl",
+              text: "#{download_url}/#{form.c[0].question.id}/audio_prompt"
+            )
+
+            assert_select(elements[1], "filename", text: "#{form.c[1].question.id}_audio_prompt.wav")
+            assert_select(elements[1], "hash", text: "ff2fd1e209465c5bffa784b5c57d84c4")
+            assert_select(
+              elements[1],
+              "downloadUrl",
+              text: "#{download_url}/#{form.c[1].question.id}/audio_prompt"
+            )
+          end
+        end
+
+        describe "should download successfully" do
+          it do
+            get("/#{audio_prompt_question_path(form.c[0].question)}")
+            expect(response).to be_success
+            expect(response.header["Content-Disposition"]).to include(form.c[0].question.id)
+          end
+
+          it do
+            get("/#{audio_prompt_question_path(form.c[1].question)}")
+            expect(response).to be_success
+            expect(response.header["Content-Disposition"]).to include(form.c[1].question.id)
+          end
```

Comment:
Very nice work here! I think this is good to go!

For checking the body, did you try `expect(response.body).to eq audio_fixture("powerup.mp3").read`?

Reply:
I did. Not in the same syntax tho and I got similar content but the special characters where different. One was using `u0003` and the other was using `x88`